NDTensors/src/lib/TypeParameterAccessors/src/Base/iswrappedarray.jl

```diff
@@ -0,0 +1,71 @@
+using SimpleTraits
+using LinearAlgebra:
+  Adjoint,
+  Diagonal,
+  Hermitian,
+  LowerTriangular,
+  Symmetric,
+  Transpose,
+  UnitLowerTriangular,
+  UnitUpperTriangular,
+  UpperTriangular
+using Base: ReshapedArray, SubArray
+using StridedViews: StridedView
+# Trait indicating if the AbstractArray type is an array wrapper.
+# Assumes that it implements `NDTensors.parenttype`.
+@traitdef IsWrappedArray{ArrayT}
+
+#! format: off
+@traitimpl IsWrappedArray{ArrayT} <- is_wrapped_array(ArrayT)
+#! format: on
+
+"""
+Optional definitions for types which are considered `Wrappers` and have a `parenttype`
+
+  Should return a `Position`.
+"""
+function parenttype_position(type::Type)
+  return 0
```

Comment:
Proposal to define a new type `struct UndefPosition end` and return `UndefPosition()` here to make it clearer when the position hasn't been defined yet, as opposed to using the convention that `0` means undefined (or non-existent).

Reply:
I think this is a great idea, this was one of my biggest concerns with the changes I made yesterday. Thanks!